api/v1alpha1/provisioning_types.go

```diff
@@ -48,6 +48,22 @@ const (
 	BootIsoSourceHttp  BootIsoSource = "http"
 )
 
+// PreprovisioningOSDownloadURLs allows specification of a set of
+// OS URLs that can be used by metal3 to pre-provision workers
+type PreprovisioningOSDownloadURLs struct {
+	// Image URL to be used for live ISO deployments
+	LiveIsoUrl string `json:"liveIsoUrl,omitempty"`
+
+	// Kernel Image URL to be used for PXE deployments
+	KernelImageUrl string `json:"kernelImageUrl,omitempty"`
```

Comment:
I would remove the Image from this name. I think having something simple like IsoUrl, KernelUrl, RootFsUrl, InitramFsUrl would be sufficient.

Reply:
Since the type is called PreprovisioningOSDownloadURLs, both "Image" and "Urls" might be unnecessary. Just calling them liveIso, kernel, initramfs, rootfs. I am open to any and all suggestions because names aren't easy.